isso/tests/test_db.py

```diff
@@ -118,3 +118,58 @@ def test_limit_nested_comments(self):
             rv = con.execute(
                 "SELECT id, parent FROM comments ORDER BY created").fetchall()
             self.assertEqual(flattened, rv)
+
+    def test_comment_text_not_null_migration(self):
+        conf = config.new({
+            "general": {
+                "dbpath": "/dev/null",
+                "max-age": "1h"
+            }
+        })
+
+        with sqlite3.connect(self.path) as con:
+            con.execute("PRAGMA user_version = 3")
+
+            con.execute("CREATE TABLE threads ("
+                        "    id INTEGER PRIMARY KEY,"
+                        "    uri VARCHAR UNIQUE,"
+                        "    title VARCHAR)")
+            con.execute("CREATE TABLE comments ("
+                        "    tid REFERENCES threads(id),"
+                        "    id INTEGER PRIMARY KEY,"
+                        "    parent INTEGER,"
+                        "    created FLOAT NOT NULL, modified FLOAT,"
+                        "    text VARCHAR, author VARCHAR, email VARCHAR, website VARCHAR,"
+                        "    mode INTEGER,"
+                        "    remote_addr VARCHAR,"
+                        "    likes INTEGER DEFAULT 0,"
+                        "    dislikes INTEGER DEFAULT 0,"
+                        "    voters BLOB NOT NULL)")
+
+            con.execute(
+                "INSERT INTO threads (uri, title) VALUES (?, ?)", ("/", "Test"))
+
+            con.execute("INSERT INTO comments (id, parent, created, text, voters) VALUES (?, ?, ?, ?, ?)",
+                        (1, None, 1, None, sqlite3.Binary(b"")))
+
+            con.execute("INSERT INTO comments (id, parent, created, text, voters) VALUES (?, ?, ?, ?, ?)",
+                        (2, 1, 2, "foo", sqlite3.Binary(b"")))
+
+        conf = config.new({
+            "general": {
+                "dbpath": "/dev/null",
+                "max-age": "1h"
+            }
+        })
+
+        db = SQLite3(self.path, conf)
+
+        self.assertEqual(db.version, SQLite3.MAX_VERSION)
+
+        with sqlite3.connect(self.path) as con:
+            # assert that the "text" field has "NOT NULL" constraint
+            rv = con.execute("SELECT \"notnull\" FROM pragma_table_info('comments') WHERE name='text'").fetchone()
```

Comment:
nit, but why `\"notnull\"` and not single-quote `'notnull'`?

Reply:
In single quotes `'notnull'` is treated as a string.

Reply:
Ah, thank you.